src/main/java/net/dv8tion/jda/internal/managers/ChannelManagerImpl.java

```diff
@@ -211,11 +230,36 @@ public ChannelManagerImpl sync(@Nonnull GuildChannel syncSource)
         Checks.notNull(syncSource, "SyncSource");
         Checks.check(getGuild().equals(syncSource.getGuild()), "Sync only works for channels of same guild");
 
-        if(syncSource.equals(getChannel()))
+        if (syncSource.equals(getChannel()))
             return this;
 
-        if (isPermissionChecksEnabled() && !getGuild().getSelfMember().hasPermission(getChannel(), Permission.MANAGE_PERMISSIONS))
-            throw new InsufficientPermissionException(getChannel(), Permission.MANAGE_PERMISSIONS);
+        if (isPermissionChecksEnabled())
+        {
+            Member selfMember = getGuild().getSelfMember();
+            if (!selfMember.hasPermission(getChannel(), Permission.MANAGE_PERMISSIONS))
+                throw new InsufficientPermissionException(getChannel(), Permission.MANAGE_PERMISSIONS);
+
+            boolean canSetRoles = selfMember.hasPermission(Permission.ADMINISTRATOR)
+                    || (PermissionUtil.getExplicitPermission(getChannel(), selfMember, false) & Permission.MANAGE_PERMISSIONS.getRawValue()) != 0;
+
+            if (!canSetRoles)
+            {
+                // Requires either administrator or MANAGE_PERMISSIONS explicitly on the channel itself
+                long botPerms = PermissionUtil.getEffectivePermission(getChannel(), selfMember) & ~Permission.MANAGE_PERMISSIONS.getRawValue();
+                boolean wouldRequireManageRoles = syncSource
+                        .getPermissionOverrides()
+                        .stream()
+                        .mapToLong(p -> p.getAllowedRaw() | p.getDeniedRaw())
```

Comment:
This should check if an existing override exists, and if thats the case, what the changed bits are.